services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java

```diff
@@ -110,8 +110,15 @@ public int postText(String uri, String text) throws IOException {
             public InputStream get(String uri) throws IOException {
                 HttpGet request = new HttpGet(uri);
                 HttpResponse response = httpClient.execute(request);
+                int statusCode = response.getStatusLine().getStatusCode();
                 HttpEntity entity = response.getEntity();
-                if (entity == null) return null;
+                if (statusCode < 200 || statusCode >= 300) {
```

Comment:
Interesting that you have to check this range manually. Isn't there a [`Response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) equivalent here?

Reply:
The closest I found is a `BasicResponseHandler` which would return a `String`, but not an `InputStream`. For the next refactoring where we rework the returned media types, we'd change this code to probably use a `BasicResponseHandler`.

Reply:
Yeah, `BasicResponseHandler` seems like a good solution. Was just wondering :)